### PR TITLE
introduce grpc server pkg

### PR DESCRIPTION
**Pull request by netsi**

### CODE REVIEW

File: pkg/grpcserver/stats_logger.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+const (
+	contextKey key = 0
+)
+
+type statsLogger struct {
+	logger   log.Logger
+	settings *Settings
+}
+
+func NewStatsLogger(settings *Settings, logger log.Logger) *statsLogger {
```
[the remark this answers is not in the document]
Comment (netsi) [reply pkg/grpcserver/stats_logger.go]: good catch, will reorder the args

File: examples/grpcserver/config.dist.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+env: dev
+app_project: project
+app_family: family
+app_name: grpc_server
+
+# grpcserver module configuration
```
[the remark this answers is not in the document]
Comment (netsi) [reply examples/grpcserver/config.dist.yml:6]: good idea, changed it in a way to support multiple grpc servers and updated the example to actually have multiple servers running (one with health checks and one without)

File: pkg/grpcserver/server.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+func New(name string, definer ServiceDefiner, middlewares ...MiddlewareFactory) kernel.ModuleFactory {
+	return func(ctx context.Context, config cfg.Config, logger log.Logger) (kernel.Module, error) {
+		var (
+			err         error
+			definitions *Definitions
+		)
+		settings := &Settings{}
+		config.UnmarshalKey(fmt.Sprintf("%s.%s", grpcServerConfigKey, name), settings)
+
+		logger = logger.WithChannel(grpcServiceChannel)
```
Comment (j4k4) [inline pkg/grpcserver/server.go:52]: Do you want to have the server name in the channel?  
Or add the server name as a field?  
To be able to distinguish between different servers?

File: pkg/grpcserver/server.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
+package grpcserver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
```
Comment (j4k4) [inline pkg/grpcserver/server.go:10]: this looks like too many import groups..  
usually we have 2 groups only.. stdlib and all the rest..

Comment (j4k4) [inline pkg/grpcserver/server.go:10]: this looks like too many import groups..  
usually we have 2 groups only.. stdlib and all the rest..

File: pkg/grpcserver/server_test.go [hunk 5]
```diff
@@ -0,0 +1,174 @@
+package grpcserver_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
```
Comment (j4k4) [inline pkg/grpcserver/server_test.go:9]: Again the thing with the import groups

Comment (j4k4) [inline pkg/grpcserver/server_test.go:9]: Again the thing with the import groups

File: pkg/grpcserver/stats_logger.go [hunk 6]
```diff
@@ -0,0 +1,114 @@
+package grpcserver
+
+import (
+	"context"
+	"sync"
+
+	"google.golang.org/grpc/stats"
```
Comment (j4k4) [inline pkg/grpcserver/stats_logger.go:7]: import groups

Comment (j4k4) [inline pkg/grpcserver/stats_logger.go:7]: import groups

File: pkg/grpcserver/server.go (hunk 4 above)
Comment (netsi) [reply pkg/grpcserver/server.go:10]: my bad was using `goimports`, will fix it for all the files I introduce

File: pkg/grpcserver/server.go (hunk 3 above)
Comment (netsi) [reply pkg/grpcserver/server.go:50]: I think it would be better to have it as a field, will add it

**Review by j4k4 — APPROVED** [review]

**Pull request merged**